lib/Db/QueueContentItemMapper.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Nextcloud - ContextChat
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Anupam Kumar <kyteinsky@gmail.com>
+ * @copyright Anupam Kumar 2024
+ */
+
+declare(strict_types=1);
+
+namespace OCA\ContextChat\Db;
+
+use OCP\AppFramework\Db\QBMapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+
+/**
+ * @template-extends QBMapper<QueueContentItem>
+ */
+class QueueContentItemMapper extends QBMapper {
+	/**
+	 * @var IDBConnection $db
+	 */
+	protected $db;
+
+	public function __construct(IDBConnection $db) {
+		parent::__construct($db, 'context_chat_content_queue', QueueContentItem::class);
+	}
+
+	/**
+	 * @param int $limit
+	 * @return list<QueueContentItem>
+	 * @throws \OCP\DB\Exception
+	 */
+	public function getFromQueue(int $limit): array {
+		$qb = $this->db->getQueryBuilder();
+		$qb->select(QueueContentItem::$columns)
+			->from($this->getTableName())
+			->setMaxResults($limit);
+
+		return $this->findEntities($qb);
+	}
+
+	/**
+	 * @param QueueContentItem $item
+	 * @return void
+	 * @throws \OCP\DB\Exception
+	 */
+	public function removeFromQueue(QueueContentItem $item): void {
+		$qb = $this->db->getQueryBuilder();
+		$qb->delete($this->getTableName())
+			->where($qb->expr()->eq('id', $qb->createPositionalParameter($item->getId())))
+			->executeStatement();
+	}
+
+	/**
+	 * @param QueueContentItem $item
+	 * @return QueueContentItem
+	 * @throws \OCP\DB\Exception
+	 */
+	public function insertIntoQueue(QueueContentItem $item) : QueueContentItem {
+		$qb = $this->db->getQueryBuilder();
+		$qb->insert($this->getTableName())
+			->values([
+				'itemId' => $qb->createPositionalParameter($item->getItemId(), IQueryBuilder::PARAM_STR),
+				'appId' => $qb->createPositionalParameter($item->getAppId(), IQueryBuilder::PARAM_STR),
+				'providerId' => $qb->createPositionalParameter($item->getProviderId(), IQueryBuilder::PARAM_STR),
+				'title' => $qb->createPositionalParameter($item->getId(), IQueryBuilder::PARAM_STR),
+				'content' => $qb->createPositionalParameter($item->getContent(), IQueryBuilder::PARAM_STR),
+				'documentType' => $qb->createPositionalParameter($item->getDocumentType(), IQueryBuilder::PARAM_STR),
+				'lastModified' => $qb->createPositionalParameter($item->getLastModified(), IQueryBuilder::PARAM_DATE),
+				'users' => $qb->createPositionalParameter($item->getUsers(), IQueryBuilder::PARAM_STR_ARRAY),
+			])
+			->executeStatement();
+		$item->setId($qb->getLastInsertId());
+		return $item;
+	}
```

Comment:
This shouldn't be necessary. There's already an insert method in QBMapper

Reply:
ah forgot to remove it, thanks. I was using it to convert the users value (array <-> csv string).